components/pricing_modal/index.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {isModalOpen} from 'selectors/views/modals';
+import {closeModal} from 'actions/views/modals';
+import {GlobalState} from 'types/store';
+import {ModalIdentifiers} from 'utils/constants';
+
+import Content from './content';
+
+import './pricing_modal.scss';
+
+function PricingModal() {
+    const dispatch = useDispatch();
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.PRICING_MODAL));
```

Comment:
The preferred pattern for show/hide/exit is to:

1. use local component state for `show`
2. default that state to `true`
3. change it to `false` when `onHide` is triggered
4. Trigger the redux closing of the modal within `onExited`

The reasoning is controlling show/hide with local component state allows for a smooth transition out. Otherwise, the exit animation gets cut off and the closing experience is more abrupt. Here's a change where we had to fix t hat in the past few months on a modal and what it looked like: https://github.com/mattermost/mattermost-webapp/pull/9550/files#diff-18fba024cda58abb038e4dec4c998073d57dec7caec2c9e0f265cb30fbcba1b1R393 , and here's the pattern as mentioned in a bug ticket https://mattermost.atlassian.net/browse/MM-40315 .

Reply:
Thanks @neallred